forte/data/base_pack.py

```diff
@@ -455,32 +456,53 @@ def on_entry_creation(
             # Use the auto-inferred control component.
             c = self.__control_component
 
-        def entry_getter(cls: Entry, attr_name: str, field_type):
+        def entry_getter(cls: Entry, attr_name: str):
             """A getter function for dataclass fields of entry object.
             When the field contains ``tid``s, we will convert them to entry
             object on the fly.
 
             Args:
                 cls: An ``Entry`` class object.
                 attr_name: The name of the attribute.
-                field_type: The type of the attribute.
             """
+
             data_store_ref = (
                 cls.pack._data_store  # pylint: disable=protected-access
             )
             attr_val = data_store_ref.get_attribute(
                 tid=cls.tid, attr_name=attr_name
             )
-            if field_type in (FList, FDict):
+            entry_type = data_store_ref.get_entry_types(
```

Comment:
It is not an `entry_type`, it is the type of the attribute. Variable names also affect the quality of the code.